src/lib/components/streams/presharedkey-modal.svelte

```diff
@@ -0,0 +1,43 @@
+<script lang="ts">
+    import Modal from 'sveltestrap/src/Modal.svelte'
+    import { Button, ModalBody, ModalFooter, Input, Label, Alert } from 'sveltestrap'
+
+    export let presharedKey: string
+    let isEqual = false
+    let open = true
+
+    function onSubmit() {
+        open = !open
+        presharedKey = undefined
+    }
+
+    function onPresharedKey(event: any) {
+        isEqual = presharedKey === event.target.value
+    }
+</script>
+
+<div>
+    <Modal isOpen={open} toggle={onSubmit}>
```

Comment:
this can be closed without clicking ok by clicking next to it. Please make sure it will only be closed if inserted the correct preshared key